### DISCUSSION TITLE
Sunbird-ED release 5.2.0

### DISCUSSION THREAD

**Discussion by vinukumar-vs** [Sunbird-Ed/Community#390] [General Discussion]
Inviting all the stakeholders to effectively help us to come up with planned release scope for 5.2 from all building blocks.

| Milestone planned| Planned dates | Comments |
| -- | --| -- |
| ED Sprint 5.2.0 | 6 Feb 2023 (delayed 9 Feb 2023) | Delayed by 1 week |
| QA Regression | 9 Feb 2023 (delayed 14 Feb 2023) | Delayed by 1 week |

Sunbird ED @Krishnaj20 @vinukumar-vs
Sunbird Knowlg @Krishnaj20 @pallakartheekreddy
Sunbird Inquiry @surendrasinghs  @pallakartheekreddy
Sunbird CoKreate @mmanimegalai @vinukumar-vs
Sunbird Lern @rhwarrier @reshmi-nair
Sunbird Observ @rhwarrier @anandp504

DC's: @SanthoshVasabhaktula @vrayulu
Sunbird QA: @ashokreddy1208
@gandham-santhosh
@kameshbhr @mmanimegalai
@madhucr @alok-os
@anandp504 @vijiurs @aishwaryashikshalokam @Prateek-slokam @aks30 @varsha-sl @MeghnaSL @Prerna789 @aayushkedia95 @pkranga @pritamps @karn-nahata @ashokreddy1208 @kameshbhr @mmanimegalai @gandham-santhosh @manju4201 @rajeevsathish @chitranshu-keshav @anisha147 

https://ed.sunbird.org/use/updating-sunbird-releases/5.2.0

Hope all the dependent building blocks has updated their latest release notes in the respective BB microsites. We will be referring 

> {{BB host}/use/release-notes/latest 

to get the latest build tags from dependent BB's. If it is not followed, we will request to share the microsite release notes of which version we should take.

**Comment by vinukumar-vs** [comment]
Waiting for other BB's to confirm their latest release tags.

Sunbird Knowlg:  [@Krishnaj20 @pallakartheekreddy](https://knowlg.sunbird.org/use/release-notes/latest)
Sunbird Inquiry:  https://inquiry.sunbird.org/use/release-notes/inquiry-release-v5.4.0-latest
Sunbird Lern: https://lern.sunbird.org/use/release-notes/release-v-5.1.0
Sunbird Observ: (**old tag itself**) https://obsrv.sunbird.org/use/release-notes/release-v-5.1.0-latest

Sunbird CoKreate: No latest tag available

I request BB owners to confirm the tags.

Sunbird Knowlg @Krishnaj20 @pallakartheekreddy
Sunbird Inquiry @surendrasinghs @pallakartheekreddy
Sunbird CoKreate @mmanimegalai @vinukumar-vs 
Sunbird Lern @rhwarrier @reshmi-nair
Sunbird Observ @rhwarrier @anandp504


DC's: @SanthoshVasabhaktula @vrayulu
Sunbird QA: @ashokreddy1208
@gandham-santhosh
@kameshbhr @mmanimegalai
@madhucr @alok-os
@anandp504 @vijiurs @aishwaryashikshalokam @Prateek-slokam @aks30 @varsha-sl @MeghnaSL @Prerna789 @aayushkedia95 @pkranga @pritamps @karn-nahata @ashokreddy1208 @kameshbhr @mmanimegalai @gandham-santhosh @manju4201 @rajeevsathish @chitranshu-keshav @anisha147

**Comment by Krishnaj20** [comment]
@vinukumar-vs - here are the release notes published by some of the building blocks which can be taken up by ED.

Lern: https://lern.sunbird.org/use/release-notes/release-v-5.1.0 - the build tags are updated in the microsite
Knowlg: https://knowlg.sunbird.org/use/release-notes 
inQuiry: https://inquiry.sunbird.org/use/release-notes/inquiry-release-v5.4.0-latest 

@pallakartheekreddy - build tags are not updated in the knowlg & inquiry microsites for the latest release. Please update so that ED team can integrate these changes.

**Reply by pallakartheekreddy** [reply]
@Krishnaj20 We have updated release notes with build tags.

**Comment by vinukumar-vs** [comment]
@santhosh-tg Lets start deployment of release-5.2.0 staging regression with below BB's tags deployed 
Lern: https://lern.sunbird.org/use/release-notes/release-v-5.1.0 - the build tags are updated in the microsite
Knowlg: https://knowlg.sunbird.org/use/release-notes/release-5.3.0
inQuiry: https://inquiry.sunbird.org/use/release-notes/inquiry-release-v5.4.0-latest

Sunbird Observ: @rhwarrier @anandp504 => No release changes. Not taking any tag
Sunbird CoKreat: @mmanimegalai @vinukumar-vs => Still waiting for the sign-off. If not provided by this week, we will not take it.

@madhucr @alok-os @anandp504 @vijiurs @aishwaryashikshalokam @Prateek-slokam @aks30 @varsha-sl @MeghnaSL @Prerna789 @aayushkedia95 @pkranga @pritamps @karn-nahata @ashokreddy1208 @kameshbhr @mmanimegalai @gandham-santhosh @manju4201 @rajeevsathish @chitranshu-keshav @anisha147 @parthlawate @kiranharidas187

**Reply by santhosh-tg** [reply]
@vinukumar-vs @ashokreddy1208 The above  mentioned BB tags are deployed to staging.  @reshmi-nair  **sunbird-lms-service**  deployment is failing with error 
`{"timestamp":"2023-02-16T07:37:44.836Z","msg":"Logger configuration in conf files is deprecated and has no effect. Use a logback configuration file instead.","lname":"application","tname":"main","level":"WARN","HOSTNAME":"learner-778575dd6c-s26l9","application.home":"/home/sunbird/learner/learning-service-1.0-SNAPSHOT"}
Exception in thread "main" java.lang.NoClassDefFoundError: akka/stream/Materializer
	at play.api.inject.BuiltinModule$$anonfun$$lessinit$greater$1.apply(BuiltinModule.scala:78)
	at play.api.inject.BuiltinModule$$anonfun$$lessinit$greater$1.apply(BuiltinModule.scala:40)
	at play.api.inject.SimpleModule.bindings(Module.scala:94)
	at play.api.inject.guice.GuiceableModuleConversions$class.guice(GuiceInjectorBuilder.scala:371)
	at play.api.inject.guice.GuiceableModule$.guice(GuiceInjectorBuilder.scala:300)
	at play.api.inject.guice.GuiceableModuleConversions$$anon$3$$anonfun$guiced$2.apply(GuiceInjectorBuilder.scala:350)
	at play.api.inject.guice.GuiceableModuleConversions$$anon$3$$anonfun$guiced$2.apply(GuiceInjectorBuilder.scala:350)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
	at scala.collection.immutable.List.foreach(List.scala:392)
	at scala.collection.TraversableLike$class.map(TraversableLike.scala:234)
	at scala.collection.immutable.List.map(List.scala:296)
	at play.api.inject.guice.GuiceableModuleConversions$$anon$3.guiced(GuiceInjectorBuilder.scala:350)
	at play.api.inject.guice.GuiceableModule$$anonfun$guiced$1.apply(GuiceInjectorBuilder.scala:326)
	at play.api.inject.guice.GuiceableModule$$anonfun$guiced$1.apply(GuiceInjectorBuilder.scala:325)
	at scala.collection.TraversableLike$$anonfun$flatMap$1.apply(TraversableLike.scala:241)
	at scala.collection.TraversableLike$$anonfun$flatMap$1.apply(TraversableLike.scala:241)
	at scala.collection.immutable.List.foreach(List.scala:392)
	at scala.collection.TraversableLike$class.flatMap(TraversableLike.scala:241)
	at scala.collection.immutable.List.flatMap(List.scala:355)
	at play.api.inject.guice.GuiceableModule$.guiced(GuiceInjectorBuilder.scala:325)
	at play.api.inject.guice.GuiceBuilder.createModule(GuiceInjectorBuilder.scala:186)
	at play.api.inject.guice.GuiceApplicationBuilder.applicationModule(GuiceApplicationBuilder.scala:125)
	at play.api.inject.guice.GuiceBuilder.injector(GuiceInjectorBuilder.scala:201)
	at play.api.inject.guice.GuiceApplicationBuilder.build(GuiceApplicationBuilder.scala:156)
	at play.api.inject.guice.GuiceApplicationLoader.load(GuiceApplicationLoader.scala:22)
	at play.core.server.ProdServerStart$.start(ProdServerStart.scala:59)
	at play.core.server.ProdServerStart$.main(ProdServerStart.scala:31)
	at play.core.server.ProdServerStart.main(ProdServerStart.scala)
Caused by: java.lang.ClassNotFoundException: akka.stream.Materializer
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	... 29 more
`
Please check 

Cc: @gandham-santhosh

**Comment by ashokreddy1208** [comment]
@vijiurs @aishwaryashikshalokam @Prateek-slokam @aks30 @varsha-sl @MeghnaSL @Preethi-Sl @Prerna789 @aayushkedia95 @pkranga @pritamps @karn-nahata @sanyapawah

I request all Contributors to share Scope, Test cases along with sign-off template for 5.2 release by today EOD 

cc @chitranshu-keshav

**Reply by omkar309** [reply]
Hello @ashokreddy1208 , We're unable to login on dev server, and hence unable to test a use case, CO-16 (https://project-sunbird.atlassian.net/browse/CO-16 ). Attaching a ss of login fail. Please guide.
![sendremindererror](https://user-images.githubusercontent.com/124027239/219617925-9440f509-5fa3-4b39-b1fd-65af7ac94f05.png)

**Reply by ashokreddy1208** [reply]
@vinukumar-vs @rajeevsathish please look into DEV env login issue

**Reply by omkar309** [reply]
@vinukumar-vs @rajeevsathish any update on this issue? Developer still not able to log in.

**Reply by vinukumar-vs** [reply]
@omkar309 this discussion thread is specific to Staging release-5.2.0. I would request to keep the dev infra discussion outside of this thread.

**Reply by omkar309** [reply]
@vinukumar-vs Understood. But we faced this issue while trying to test a use case, CO-16 (https://project-sunbird.atlassian.net/browse/CO-16 ) that is part of release 5.2.0.

Please guide where to raise this issue outside of this thread.

**Comment by sanyapawah** [comment]
Adding Abhay and Omkar.

On Fri, 17 Feb 2023 at 9:55 AM, ashokreddy1208 ***@***.***>
wrote:

> @vijiurs <https://github.com/vijiurs> @aishwaryashikshalokam
> <https://github.com/aishwaryashikshalokam> @Prateek-slokam
> <https://github.com/Prateek-slokam> @aks30 <https://github.com/aks30>
> @varsha-sl <https://github.com/varsha-sl> @MeghnaSL
> <https://github.com/MeghnaSL> @Preethi-Sl <https://github.com/Preethi-Sl>
> @Prerna789 <https://github.com/Prerna789> @aayushkedia95
> <https://github.com/aayushkedia95> @pkranga <https://github.com/pkranga>
> @pritamps <https://github.com/pritamps> @karn-nahata
> <https://github.com/karn-nahata> @sanyapawah
> <https://github.com/sanyapawah>
>
> I request all Contributors to share Scope, Test cases along with sign-off
> template for 5.2 release by today EOD
>
> cc @chitranshu-keshav <https://github.com/chitranshu-keshav>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/Sunbird-Ed/discussions/390#discussioncomment-5004091>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A25FHDMQUI3XJWA5F7RBCR3WX34T7ANCNFSM6AAAAAAUU6Q244>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
-- 
Thanks,

*Sanya Pawah*

Senior Associate | 9549060608

Samagra | Transforming Governance

Address: 9/5 Sarvapriya Vihar, New Delhi-16
***@***.***,77.1994501,17z/data=!3m1!4b1!4m5!3m4!1s0x390ce273981e5dfb:0x7bfceb198af17562!8m2!3d28.5566128!4d77.2016388?shorturl=1>

Website: www.samagragovernance.in

Follow Us: Linkedin
<https://www.linkedin.com/company/samagra-transforming-governance/> |
Youtube <https://www.youtube.com/channel/UCfkXErS-f87xUQkmSKSC8bg> |
Instagram <https://www.instagram.com/samagragovernance/?hl=en>

**Comment by ashokreddy1208** [comment]
@gandham-santhosh Can you please confirm on 5.2 Deployment status on sunbird ED staging environment.

@chitranshu-keshav We are waiting for a confirmation to start 5.2 regression.

cc 
Sunbird ED @vinukumar-vs @Krishnaj20 
Sunbird Knowlg @Krishnaj20 @pallakartheekreddy
Sunbird Inquiry @surendrasinghs @pallakartheekreddy
Sunbird CoKreate @mmanimegalai @vinukumar-vs
Sunbird Lern @rhwarrier @reshmi-nair
Sunbird Observ @rhwarrier @anandp504

DC's: @SanthoshVasabhaktula @vrayulu
Dev ops: @gandham-santhosh
@kameshbhr @mmanimegalai
@madhucr @alok-os
@anandp504 @vijiurs @aishwaryashikshalokam @Prateek-slokam @aks30 @varsha-sl @MeghnaSL @Prerna789 @aayushkedia95 @pkranga @pritamps @karn-nahata @manju4201 @rajeevsathish @anisha147 @omkar309 @sanyapawah @vijiurs @aishwaryashikshalokam @Prateek-slokam @aks30 @varsha-sl @MeghnaSL @Preethi-Sl @Prerna789 @aayushkedia95 @pkranga @pritamps @karn-nahata @sanyapawah

**Reply by gandham-santhosh** [reply]
@ashokreddy1208 there is an issue with lms service deployment. Checking the same with Lern team
@reshmi-nair can you please check this. Please refer the comment by @santhosh-tg

**Comment by ashokreddy1208** [comment]
@reshmi-nair we will wait for your confirmation to start 5.2 regression

cc @vinukumar-vs @chitranshu-keshav @gandham-santhosh @Krishnaj20

**Reply by Ak0094** [reply]
Changes in lern 5.1 are not available on the sunbird environment and we are blocked to test the new features.
cc: @vinukumar-vs  @reshmi-nair @rhwarrier @ashokreddy1208 @gandham-santhosh

**Reply by reshmi-nair** [reply]
@ashokreddy1208 @Ak0094 Lern 5.1 changes are deployed. Pls check now.
cc: @vinukumar-vs @rhwarrier @gandham-santhosh

**Comment by Priyakashi** [comment]
Changes in Knowlg 5.3 is not available on staging

@vinukumar-vs @ashokreddy1208  [@Santhosh Gandham](https://ekstep.slack.com/team/U78TDJSD7) [@kartheek](https://ekstep.slack.com/team/U1U63B5P1)

**Reply by vinukumar-vs** [reply]
@gandham-santhosh I hope Infra configuration changes & services are deployed for Knowlg. Please confirm

@pallakartheekreddy @Krishnaj20

**Comment by Aparna-karri591** [comment]
Inquiry R.5.1,R5.2 ,R5.3and R5.4 changes are not available on staging

@vinukumar-vs @ashokreddy1208 @gandham-santhosh @pallakartheekreddy

**Reply by Aparna-karri591** [reply]
@vinukumar-vs as discussed Inquiry BB deployment is not done for R 5.3 and R5.4 of Inquiry to staging , Request you to please deploy same on sunbird staging environment and confirm us back to start 5.2 Regression

CC:  @ashokreddy1208 @gandham-santhosh @pallakartheekreddy @chitranshu-keshav  @Krishnaj20

**Reply by vinukumar-vs** [reply]
@pallakartheekreddy @Aparna-karri591 @Krishnaj20 
InQuiry building block changes we can't take it because of front-end npm modules are upgraded to Angular-12 & it is not documented  anywhere in the release notes.
https://inquiry.sunbird.org/use/release-notes/inquiry-release-v5.4.0-latest

> Release Tags:
Question Set Editor:
Tag: v5.4.0
Install: npm i @project-sunbird/sunbird-collection-editor@5.4.0
URL: https://www.npmjs.com/package/@project-sunbird/sunbird-collection-editor/v/5.4.0

> 
> Question Set Player:
> Tag: v5.4.0
> Install: npm i @project-sunbird/sunbird-quml-player@5.4.0
> URL: https://www.npmjs.com/package/@project-sunbird/sunbird-quml-player/v/5.4.0

@ashokreddy1208 @chitranshu-keshav @vrayulu @SanthoshVasabhaktula @surendrasinghs

**Reply by pallakartheekreddy** [reply]
@vinukumar-vs inQuiry BB upgraded to Angular-12 in release-5.1 not in 5.4. Please refer 5.1 release notes where we added `Deprecation Warning`
https://inquiry.sunbird.org/use/release-notes/inquiry-release-v5.1.0
@Krishnaj20 @surendrasinghs

**Reply by vinukumar-vs** [reply]
@surendrasinghs @pallakartheekreddy We can't take the player & editor changes untill we upgrade to 12. None of the Inquiry changes will be available in ED till Next major release-6.0.0

**Reply by vinukumar-vs** [reply]
> @surendrasinghs @pallakartheekreddy We can't take the player & editor changes untill we upgrade to 12. None of the Inquiry changes will be available in ED till Next major release-6.0.0

@Krishnaj20 @chitranshu-keshav

**Reply by vinukumar-vs** [reply]
ED has no taken Inquiry release-5.4.0 
**reason:** 
https://github.com/Sunbird-inQuiry/Community/discussions/63

@rajeevsathish @Krishnaj20 @ashokreddy1208

**Comment by ashokreddy1208** [comment]
@chitranshu-keshav We are starting 5.2 Regression on sunbird ED env from today.

Note :  We didn't receive any update for contributors on 5.2 release till time.

cc
Sunbird ED @vinukumar-vs @Krishnaj20
Sunbird Knowlg @Krishnaj20 @pallakartheekreddy
Sunbird Inquiry @surendrasinghs @pallakartheekreddy
Sunbird CoKreate @mmanimegalai @vinukumar-vs
Sunbird Lern @rhwarrier @reshmi-nair
Sunbird Observ @rhwarrier @anandp504

DC's: @SanthoshVasabhaktula @vrayulu
Dev ops: @gandham-santhosh
@kameshbhr @mmanimegalai
@madhucr @alok-os
@anandp504 @vijiurs @aishwaryashikshalokam @Prateek-slokam @aks30 @varsha-sl @MeghnaSL @Prerna789 @aayushkedia95 @pkranga @pritamps @karn-nahata @manju4201 @rajeevsathish @anisha147 @omkar309 @sanyapawah @vijiurs @aishwaryashikshalokam @Prateek-slokam @aks30 @varsha-sl @MeghnaSL @Preethi-Sl @Prerna789 @aayushkedia95 @pkranga @pritamps @karn-nahata @sanyapawah

**Comment by vinukumar-vs** [comment]
@gandham-santhosh  Lets do the deployment of hot-fix of Knowlg 5.2.1 into Sunbird-ED staging for the below knowlg jobs
**Sunbird Knowlg**(hot-fix): [Release - 5.2.1](https://knowlg.sunbird.org/use/release-notes/release-5.2.0-ongoing#hot-fix-5.2.1-20-01-2023) 

- **csp-migrator** Flink job
- **content-publish** Flink job

Hot-fix Issue: https://project-sunbird.atlassian.net/browse/ED-1617

This is already fixed by knowlg as hot-fix but we have not taken this as part of release-5.2.0(communication gap). Hence we have to release as hot-fix of release-5.2.0 of Sunbird-ED.

@Krishnaj20 @chitranshu-keshav @ashokreddy1208 @anisha147

**Reply by gandham-santhosh** [reply]
@vinukumar-vs this is deployed to Sunbird-Ed staging environment

**Comment by vinukumar-vs** [comment]
ED-5.2.0 releases

- **Sunbird Knowlg(hot-fix)** - [Release - 5.2.1](https://knowlg.sunbird.org/use/release-notes/release-5.2.0-ongoing#hot-fix-5.2.1-20-01-2023)
- Sunbird Lern: https://lern.sunbird.org/use/release-notes/release-v-5.1.0 
- Sunbird Knowlg: https://knowlg.sunbird.org/use/release-notes/release-5.3.0
- Sunbird inQuiry: https://inquiry.sunbird.org/use/release-notes/inquiry-release-v5.3.0
- Sunbird Obsrv -  https://obsrv.sunbird.org/use/release-notes/release-v-5.1.0-latest
- Sunbird CoKreat - https://cokreat.sunbird.org/use/release-notes/cokreat-release-v5.2.0

@gandham-santhosh Can you confirm the tags. So that the next release, we can follow upgradation of services release by release

@Krishnaj20 @ashokreddy1208 @rajeevsathish @chitranshu-keshav

**Comment by chitranshu-keshav** [comment]
Since Release 5.2.0 is complete, closing this thread
